api/applications_api/serializers.py

```diff
@@ -686,3 +688,49 @@ def create(self, validated_data):
 
         instance.save()
         return instance
+
+
+class ApplicationEventWeeklyAmountReductionSerializer(serializers.ModelSerializer):
+    application_event_schedule_result_id = serializers.PrimaryKeyRelatedField(
+        source="application_event_schedule_result",
+        queryset=ApplicationEventScheduleResult.objects.all(),
+    )
+    application_event_id = serializers.PrimaryKeyRelatedField(
+        source="application_event", read_only=True
+    )
+    user = serializers.HiddenField(default=NullableCurrentUserDefault())
+
+    class Meta:
+        model = ApplicationEventWeeklyAmountReduction
+        fields = [
+            "id",
+            "application_event_schedule_result_id",
+            "user",
+            "application_event_id",
+        ]
+
+    def create(self, validated_data):
+        schedule_result = validated_data.pop("application_event_schedule_result")
+
+        schedule_result.delete()
+        reduction = super().create(validated_data)
+        return reduction
+
+    def validate(self, data):
+        result = data["application_event_schedule_result"]
+
+        # TODO: add validation for approved when other pr merged.
```

Comment:
Should this TODO be done now? Or is that validation the lower `if result.accepted`

Reply:
Oh yeah, it's done. Should remove the todo.